### DISCUSSION TITLE
Enabling Charged Species in RMG-py

### DISCUSSION THREAD

**Discussion by sptiwari** [ReactionMechanismGenerator/RMG-Py#2738] [Q&A] [answered]
RMG-py currently restricts the inclusion of charged species. How can I enable reaction mechanism generation for such species? Specifically, what modifications are required in the `input.py` file, or are there changes needed in the source code? Additionally, what assumptions and potential risks should I be aware of when enabling this functionality?

I plan to create my own kinetics library for the reaction of an amine with CO2 and water.

**Comment by JacksonBurns** [comment, marked as answer]
Hi @sptiwari it is very difficult to add support for ions.

You would need to follow the example of adding Li+ done here: https://github.com/ReactionMechanismGenerator/RMG-Py/pull/2598 and here: https://github.com/ReactionMechanismGenerator/RMG-database/pull/667
This involves adding a new atom type, adding thermo and kinetics estimators, etc.

As you can see from these PRs this a multi-year long process

**Comment by sptiwari** [comment]
Thank you for the pointers. I will check these links.
